### PR TITLE
Localization refactor

### PR DESCRIPTION
**Pull request by vergilius**
Screenshot 2019-09-10 at 12 05 46
Screenshot 2019-09-10 at 12 05 52

### CODE REVIEW

File: src/AppContainer.jsx [hunk 1]
```diff
@@ -161,26 +163,37 @@ class AppContainer extends React.Component<PropsType, StateType> {
 
   render() {
     const {data} = this.state;
+    const {getRoute} = this.props;
     const screenType = widthToScreenType(this.state.width);
     const isWideLayout = screenType === SCREEN_DESKTOP;
 
     return (
       <>
+        <div style={{
```
Comment (vergilius) [inline src/AppContainer.jsx:172]: just a debug

File: src/components/navigation/Navigation.jsx [hunk 2]
```diff
@@ -99,4 +102,4 @@ class Navigation extends React.Component<*, *> {
   }
 }
 
-export default withRouter(Navigation);
+export default withLocale()<*>(withRouter(Navigation));
```
Comment (vergilius) [inline src/components/navigation/Navigation.jsx:105]: I need to fix flow types later globally ;(

File: src/components/navigation/NavigationDesktop.jsx [hunk 3]
```diff
@@ -38,40 +44,41 @@ class Navigation extends React.Component<*, *> {
...
-            to="/info"
-            className={classnames(style.navigationLink, {
-              [`${style.navigationLinkDetail}`]: isPresent || isPast,
-            })}
+            to={generateRoute('CHANGE_LANGUAGE')}
+            className={linkClassName}
+            onClick={changeLocale}
           >
-            Info
+            {locale === 'pl' ? '🇺🇸' : '🇵🇱'}
```
Comment (vergilius) [inline src/components/navigation/NavigationDesktop.jsx:58]: just an example

File: src/utils/routes/router.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+// @flow strict
+
+import routes from './routes';
+
+function getRoute(id: string, locale: 'en' | 'pl') {
```
Comment (vergilius) [inline src/utils/routes/router.js:5]: returns basic path for react router

File: src/utils/routes/router.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  return routes[locale][id];
+}
+
+function generateRoute(id, params: Object | null, locale: 'en' | 'pl') {
```
Comment (vergilius) [inline src/utils/routes/router.js:9]: returns generated path for links (replaces `:attr` with given params)

File: src/utils/routes/router.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  const route = getRoute(id, locale);
+
+  if (params) {
+    return route
```
Comment (vergilius) [inline src/utils/routes/router.js:13]: this might be probably written in a better way, atm it does its job

File: src/utils/url.js [hunk 7]
```diff
@@ -1,4 +1,8 @@
 // @flow strict
 
 export const slugifyStreetName = (name: string) =>
-  name.toLowerCase().replace(' ', '-');
+  name
```
Comment (vergilius) [inline src/utils/url.js:4]: this has impact on how URL's are transformed, see #39 for details

File: src/components/navigation/Navigation.jsx (hunk 2 above)
Comment (burakukula) [reply src/components/navigation/Navigation.jsx:105]: sounds like so much fun

**Review by burakukula — APPROVED** [review]
OMG, Micz, amazing work ❤️

**Pull request merged**